packages/hash/frontend/src/blocks/page/createSuggester/createSuggester.tsx

```diff
@@ -121,35 +137,57 @@ export const createSuggester = (
       },
       /** produces a new state from the old state and incoming transactions (cf. reducer) */
       apply(tr, state, _prevEditorState, nextEditorState) {
-        const action: SuggesterAction | undefined = tr.getMeta(key);
+        const action: SuggesterAction | undefined =
+          tr.getMeta(suggesterPluginKey);
 
         switch (action?.type) {
           case "escape":
-            return { ...state, disabled: true };
+            return { ...state, disabled: true, suggestedBlockPosition: null };
+
+          case "key":
+            return { ...state, suggestedBlockPosition: null };
+
+          case "suggestedBlock":
+            return {
+              ...state,
+              suggestedBlockPosition: action.payload.position,
+            };
         }
 
+        const suggestedBlockPosition =
+          state.suggestedBlockPosition === null ||
+          tr.selectionSet ||
+          docChangedInTransaction(tr)
+            ? null
+            : tr.mapping.map(state.suggestedBlockPosition);
```

Comment:
Can we add a comment explaining this please – either inline or in the docstring on the function

Reply:
[`5177fad` (#872)](https://github.com/hashintel/hash/pull/872/commits/5177fad45ecc38b16a6d6d04ed0decf7d722dc45)

Reply:
Thank you!